src/Tools/babylon.tools.ts

```diff
@@ -395,7 +395,9 @@
 
                 request.onreadystatechange = () => {
                     if (request.readyState === 4) {
-                        if (request.status === 200 || Tools.ValidateXHRData(request, !useArrayBuffer ? 1 : 6)) {
+                        request.onreadystatechange = null;//some browsers have issues where onreadystatechange can be called multiple times with the same value
+
+                        if (request.status >= 200 && request.status < 300) {

```

Comment:
Would it not be better to handle the special case for Cocoon (ie `|| (navigator.isCocoonJS && (request.status === 0))`)? At present it's broken in all browsers